core/src/main/java/io/paradoxical/cassieq/auth/SignedRequestAuthFilter.java

```diff
@@ -0,0 +1,171 @@
+            if (pathParameters.containsKey(accountNamePathParameter)) {
+
+                final AccountName accountName = AccountName.valueOf(pathParameters.getFirst(accountNamePathParameter));
+
+                final RequestParameters headerRequestParameters = parseHeaderRequestParameters(requestContext, accountName);
+                final SignedUrlParameters signedUrlParameters = parseSignedUrlParameters(requestContext, accountName);
+
+                final AuthorizedRequestCredentials credentials =
+                        AuthorizedRequestCredentials.builder()
+                                                    .requestParameters(MoreObjects.firstNonNull(headerRequestParameters, signedUrlParameters))
+                                                    .build();
+
+                if (setPrincipal(requestContext, credentials)) {
+                    return;
+                }
+            }
+            else {
+                return;
+            }
+        }
+        catch (IllegalArgumentException e) {
+            logger.warn("Error decoding credentials", e);
+        }
+
+        throw new WebApplicationException(unauthorizedHandler.buildResponse(prefix, realm));
+    }
+
+    private RequestParameters parseHeaderRequestParameters(final ContainerRequestContext requestContext, final AccountName accountName) {
+        final String header = requestContext.getHeaders().getFirst(HttpHeaders.AUTHORIZATION);
+
+        if (!Strings.isNullOrEmpty(header)) {
+            final Splitter splitter = Splitter.on(' ').limit(2).trimResults();
+            final List<String> components = splitter.splitToList(header);
+
+            if (components.size() == 2) {
+                final String authScheme = components.get(0);
+
+                if (keyAuthPrefix.equalsIgnoreCase(authScheme)) {
+                    return AccountKeyParameters.builder()
+                                               .accountName(accountName)
+                                               .authKey(components.get(1))
+                                               .build();
+                }
+
+                if (prefix.equalsIgnoreCase(authScheme)) {
+                    return SignedRequestParameters.builder()
+                                                  .accountName(accountName)
+                                                  .requestMethod(requestContext.getMethod())
+                                                  .providedSignature(components.get(1))
+                                                  .requestPath(requestContext.getUriInfo().getPath())
+                                                  .build();
+                }
+            }
+        }
+
+        return null;
+    }
+
+    private SignedUrlParameters parseSignedUrlParameters(final ContainerRequestContext requestContext, final AccountName accountName) {
+        final String sig = requestContext.getUriInfo().getQueryParameters().getFirst("sig");
```

Comment:
Probably ought to centralize these query param values, cause we'll want to re-use these same params when we generate query param auth later